src/components/Product.js

```diff
@@ -0,0 +1,25 @@
+import MoneyButton from "./MoneyButton";
+import ProgressBar from "./ProgressBar";
+import useStore from "~/src/zustand/store";
+import { useState } from "react";
+
+export default function Product() {
+  const { addMoney, tier } = useStore();
+  const [isFilling, changeFilling] = useState(false);
```

Comment:
a more common naming would be:
```suggestion
  const [isFilling, setIsFilling] = useState(false);
```